one-ui/ui/src/components/flows/flows.tsx

```diff
@@ -0,0 +1,283 @@
+    }
+
+    let panels;
+    if (props.flows) {
+        panels = props.flows.map((flow, i) => {
+            let name = flow.name;
+            let indexes = Object.keys(flow.steps);
+            let cards = indexes.map((i) => {
+                let step = flow.steps[i];
+                // TODO Handle steps that don't have input formats
+                let stepFormat = (step.fileLocations) ?  getLoadSourceFormat(step.name) : 'json';
+                return (
+                    <Card 
+                        style={{ width: 300, marginRight: 20 }} 
+                        title={StepDefToTitle(step.stepDefinitionType)} 
+                        size="small"
+                        extra={
+                            <div className={styles.actions}>
+                                <div className={styles.run} onClick={() => props.runStep(step.name)}><Icon type="play-circle" theme="filled" /></div>
+                                {props.canWriteFlows ?
+                                    <Tooltip 
+                                        title={'Delete Step'} 
+                                        placement="bottom"
+                                    >
+                                        <div className={styles.delete} onClick={() => handleStepDelete(flow.name, step.name)}><Icon type="close" /></div>
+                                    </Tooltip> :
+                                    <Tooltip 
+                                        title={'Delete Step'} 
+                                        placement="bottom"
+                                    >
+                                        <div className={styles.disabledDelete} onClick={(event) => { event.stopPropagation(); event.preventDefault(); }}><Icon type="close" /></div>
+                                    </Tooltip> 
+                                }
+                            </div>
+                        }
+                    >
+                        <div className={styles.cardContent}>
+                            <div className={styles.format} style={sourceFormatStyle(stepFormat)}>{stepFormat.toUpperCase()}</div>
+                            <div className={styles.name}>{step.name}</div>
+                        </div>
+                    </Card>
+                )
+            });
+            return (
+                <Panel 
+                    header={flowHeader(flow.name, i)} 
+                    key={i} 
+                    extra={deleteIcon(name)}
+                >
+                    <div className={styles.panelContent}>
+                        {cards}
+                    </div>
+                </Panel>
+            )
+        })
+    }
+
+   return (
+    <div id="flows-container" className={styles.flowsContainer}>
+        {props.canReadFlows || props.canWriteFlows ?
+            <>
+                <div className={styles.createContainer}>
+                    <Button 
```

Comment:
@wooldridge should this be MLButton?
The edit feature should be similar to load data config. An operator should be able to click open the "Edit Flow" dialog to view.
Just for readability, I am not too particular about this. Can properties that belong to one element be in one line? Its ok when there is only 1 or 2 elements. When there are more like in this return block to me its more readable if they are in one line.

Reply:
Added MlButton and committed. Let's move the discussion of general coding style out to the broader group (including the Explorer team).